__tests__/commands/cache.js

```diff
@@ -0,0 +1,65 @@
+/* @flow */
+
+import * as reporters from '../../src/reporters/index.js';
+import * as fs from '../../src/util/fs.js';
+import {run} from '../../src/cli/commands/cache.js';
+import {run as buildRun, runInstall} from './_helpers.js';
+
+const path = require('path');
+const stream = require('stream');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'cache');
+
+const runCache = buildRun.bind(
+  null,
+  reporters.ConsoleReporter,
+  fixturesLoc,
+  async (args, flags, config, reporter, lockfile, getStdout): Promise<string> => {
+    await run(config, reporter, flags, args);
+    return getStdout();
+  },
+);
+
+test('ls', async (): Promise<void> => {
+  await runInstall({}, 'artifacts-finds-and-saves', async (config): Promise<void> => {
+    const out = new stream.PassThrough();
+    const reporter = new reporters.ConsoleReporter({stdout: out});
+    await run(config, reporter, {}, ['ls']);
+    const stdout = String(out.read());
```

Comment:
Wouldn't it be easier and more reliable to use the JSON reporter and check its output?